### PR TITLE
fix(release): attempt to delete stray RCs when Release DB model does a delete

### PR DESCRIPTION
**Pull request by helgi**
# Summary of Changes

This will reduce the occurance of stray RCs when the Release record is nuked, due to whatever problems may arise

# Issue(s) that this PR Closes

I believe this will help with #319 and possibly others

# Testing Instructions

Somewhat hard to test this one since it is not possible to trigger a Release model delete without causing havoc in other places, which we also try to prevent

### CODE REVIEW

File: rootfs/api/models/release.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def publish(self, source_version='latest'):
             source_image.startswith(settings.REGISTRY_HOST) or
             source_image.startswith(settings.REGISTRY_URL)
         ):
-            logger.debug('{} already exists in the target registry. Using this image for release {} of app {}'.format(source_image, self.version, self.app))  # noqa
+            log_event(self.app, '{} already exists in the target registry. Using this image for release {} of app {}'.format(source_image, self.version, self.app))  # noqa
```
Comment (arschles) [inline rootfs/api/models/release.py]: is this related to the intent of the PR? ok if it is, just wondering for context

File: rootfs/api/models/release.py [hunk 2]
```diff
@@ -148,6 +148,27 @@ def rollback(self, user, version):
...
+                self._scheduler._delete_rc(self.app.id, controller['metadata']['name'])
+        except KubeHTTPException as e:
+            # 404 means they were already cleaned up
+            if e.status_code is not 404:
+                # Another problem came up
+                message = 'Could not to cleanup RCs for release {}'.format(self.version)
+                log_event(self.app, message)
+                logger.warning(message + ' - ' + str(e))
+
+        super(Release, self).delete(*args, **kwargs)
```
Comment (arschles) [inline rootfs/api/models/release.py]: would it make sense to call this in a `finally` block, so it's called no matter what?

File: rootfs/api/models/release.py (hunk 1 above)
Comment (helgi) [inline rootfs/api/models/release.py]: Not directly related beyond I realised I was relaying information in the wrong context and made this match what I did in the delete exception

File: rootfs/api/models/release.py (hunk 2 above)
Comment (helgi) [inline rootfs/api/models/release.py]: The codebase does not have many (any?) `finally` but it feels like a good practice to do since we want to delete no matter what, added it in

File: rootfs/api/models/release.py (hunk 1 above)
Comment (arschles) [inline rootfs/api/models/release.py]: sgtm
